options.go

```diff
@@ -0,0 +1,29 @@
+package goSam
+
+import (
+	"strconv"
+)
+
+type Option func(*Client) error
+
+func SetAddr(s string) func(*Client) error {
+	return func(c *Client) error {
+		c.addr = s
+		return nil
+	}
+}
+
+func SetPort(s string) func(*Client) error {
+	return func(c *Client) error {
+		port, err := strconv.Atoi(s)
+		if err != nil {
+			return err
+		}
+		if port < 65536 && port > -1 {
```

Comment:
I am ready to make this change, but should the error indicate that the reason for the invalid port is the conversion from string to integer? Like, "Invalid port; non-number."

Reply:
oh sorry. i misread this.

i glanced over the negative `if` because i'm more used to code like this:

```go
if os.IsEOF(err) {
  // cleanup or log
  return err
}
// go on
```

than 

```go
if _what i want_ {
  // go on
} else {
   // cleanup or log
  return err
}

```

Reply:
In the upper case i prefer to use pkg/errors to wrap it like `errors.Wrap(err, "port: invalid input port")` then you also see the strconv reason with context.